src/utils/mergeClasses.ts

```diff
@@ -0,0 +1,40 @@
+import { ScrollbarsProps, StyleClasses } from '..';
+import classNames from 'classNames';
+
+function getDynamicClasses(props: ScrollbarsProps): Partial<StyleClasses> {
+  return {
+    root: classNames({
+      'rc-scrollbars-heightAuto': props.autoHeight,
+    }),
+  };
+}
+
+export default function getFinalClasses(props: ScrollbarsProps): StyleClasses {
+  const { className, classes } = props;
+  const dynamicClasses = getDynamicClasses(props);
+
+  return {
+    root: classNames('rc-scrollbars-container', className, dynamicClasses.root, classes.root),
+    view: classNames('rc-scrollbars-view', dynamicClasses.view, classes.view),
+    trackVertical: classNames(
+      'rc-scrollbars-track',
+      dynamicClasses.trackVertical,
+      classes.trackVertical,
+    ),
+    trackHorizontal: classNames(
+      'rc-scrollbars-track',
+      dynamicClasses.trackHorizontal,
+      classes.trackHorizontal,
+    ),
+    thumbVertical: classNames(
+      'rc-scrollbars-thumb',
+      dynamicClasses.thumbVertical,
+      classes.thumbVertical,
+    ),
+    thumbHorizontal: classNames(
+      'rc-scrollbars-thumb',
+      dynamicClasses.thumbHorizontal,
+      classes.thumbHorizontal,
+    ),
+  };
+}
```

Comment:
My opinion that "classnames" is useless here. 
Also, getFinalClasses doesn't resolve the merge problem. Projects, where are thumbs customized by render props with className, will lose their styles. 

1 feature will break down another.

Reply:
The intention here was the same as on the root level - to merge all 3 class string. 
Point that moving styles from the styles object is a breaking change is valid! Damn, and I wanted to retract from inline styles as much as possible (leave only the dynamic values as a necessity).